### DISCUSSION TITLE
Question about estimate_depletion_voltage() and depletion voltage in the experiment

### DISCUSSION THREAD

**Discussion by zql2021** [JuliaPhysics/SolidStateDetectors.jl#373] [Q&A] [answered]
Dear Sir,
I'm encountering an issue where the actual depletion voltage of my detector during experiment is around 300V, but the result calculated using **estimate_depletion_voltage(sim)** is 1.2634277f0 V. This discrepancy seems too significant. I wonder if the depletion voltage calculated by estimate_depletion_voltage() represents the same concept as the depletion voltage in the experiment. Could you provide some insights on how to check and debug my simulation code? 
Thank you for your assistance.

**Comment by fhagemann** [comment]
It's a bit hard to tell what exactly the problem is without knowing the detector geometry you're using.
Could you try `SolidStateDetectors.old_estimate_depletion_voltage(sim)` and see whether that result is also far away from your expectations?

**Comment by fhagemann** [comment, marked as answer]
One thing that might help debugging:
if you run your simulations for a given depletion voltage under the assumed depletion voltage (make sure to use `depletion_handling = true`), you can check if it's depleted using `is_depleted(sim.point_types)`. Plotting the `point_types` might help to find out where exactly your detector is still undepleted (search for yellow areas).

Another thing would be to check how you are defining the impurity density in the config file:
- sometimes units get a bit mixed up in the config files. Are you sure that you are passing the impurity density values in the correct units? To be 100% sure, you can also explicitly add the units in the config file, e.g. [here](https://github.com/JuliaPhysics/SolidStateDetectors.jl/blob/e3d9dbf5b9497bd131bad323dcd1b9c2bbd45b2e/examples/example_config_files/isochrone_test.yaml#L29-L31)
```
impurity_density:
   name: "constant"
   value: 1e7cm^-3  # for 10^7 cm^(-3)
```
- Are you passing the impurity density with the correct sign (positive for n-type, negative for p-type)?

**Reply by zql2021** [reply]
Oh,the problem has been solved！it was caused by the sign of the impurity density. Thank you very much!
